src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp

```diff
@@ -328,6 +290,146 @@ class boss_ichoron : public CreatureScript
+                    {
+                        bIsExploded = true;
+                        Talk(SAY_SHATTER);
+                        DoCast(SPELL_BURST);
+                        me->RemoveAllAuras();
+                        burstTimer = 15000;
+
+                        std::list<Creature*> summonTargets;
+                        GetCreatureListWithEntryInGrid(summonTargets, me, NPC_ICHORON_SUMMON_TARGET, 200.0f);
+                        std::list<Creature*>::iterator itr = summonTargets.begin();
+                        for (uint8 i = 0; i < 10; i++)
+                        {
+                            std::advance(itr, urand(0, summonTargets.size() - 1));
+                            Position targetPos = (*itr)->GetRandomNearPosition(10.0f);
+                            itr = summonTargets.begin();
+                            TempSummon* globule = me->SummonCreature(NPC_ICHOR_GLOBULE, targetPos, TEMPSUMMON_CORPSE_DESPAWN);
+                            DoCast(globule, SPELL_WATER_GLOBULE_VISUAL);
+
+                            float minDistance = 1000;
+                            uint8 k = 0;
+                            for (uint8 j = 0; j < 10; j += 2)
+                            {
+                                if (globule->GetDistance(globulePaths[j]) < minDistance)
+                                {
+                                    minDistance = globule->GetDistance(globulePaths[j]);
+                                    k = j;
+                                }
+                            }
+
+                            globule->GetAI()->SetData(0, k);
+                        }
+                        return;
+                    }
+
+                    if (me->HasUnitState(UNIT_STATE_CASTING))
+                        return;
+
+                    events.Update(uiDiff);
+
+                    switch (uint32 eventId = events.ExecuteEvent())
+                    {
+                    case EVENT_WATER_BLAST:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+                            DoCast(target, SPELL_WATER_BLAST);
+                        events.ScheduleEvent(EVENT_WATER_BLAST, urand(6000, 9000));
+                        break;
+                    case EVENT_WATER_BOLT_VOLLEY:
+                        DoCast(SPELL_WATER_BOLT_VOLLEY);
+                        events.ScheduleEvent(EVENT_WATER_BOLT_VOLLEY, urand(10000, 15000));
+                        break;
+                    }
+
+                    DoMeleeAttackIfReady();
+                }
+                else if (!bIsDrained)
+                {
+                    if (drainedTimer <= 0)
+                    {
+                        bIsDrained = true;
+                        drainedTimer = 50;
+                        uint32 damage = me->CountPctFromMaxHealth(30);
+                        if (me->GetHealth() < damage)
+                        {
+                            uint32 Hp1 = me->GetMaxHealth() * (0.01);
```

Comment:
please watch out for implicit casts between double and uint32, you might also want to just do 
```
me->GetMaxHealth() / 100;
```
to achieve same result and avoid any type casting